### PR TITLE
Sav fix lead team of x badge

### PR DESCRIPTION
**Pull request by Lunaliar**
# Description
This bug resulted from improper backend calls and a typo in the aggregation pipeline of the database for the myTeam collection and null values in backend permissions of several roles. I have removed permissions from the function so it will run faster and refactored the function. This badge should be awarded to users with either the mentor or the manager role and will only count users without the roles of Admin, Owner, Coreteam, manager, or mentor as team members being led. For example, if your user was on a team with 10 users with leadership roles and 5 volunteers they would be awarded a badge for leading a team of 5 users. To test this function make a user a mentor or manager and then add them to a team of at least 5 volunteers to make sure this badge is being awarded properly. Best of luck please watch the video for more clarification.

## Main changes explained:
Fix getTeamMembers function to call database properly
Remove hasPermission calls.
Refactor code to run faster.
…

## How to test:
check into current branch
log as admin user
Make a mentor/manager user
Add user to team with 5 or more volunteers 
isolate tested mentor/manager user in awardBadges
## Screenshots or videos of changes:
https://github.com/OneCommunityGlobal/HGNRest/assets/98275089/9bb0693a-948b-47af-9815-c1dae67d2087

### CODE REVIEW

**Review by 20chen-7 — APPROVED** [review]
The function is intended. Everything looks good. This is some error in the terminal after the test. I am not sure that is another error that may be required to solve in the future.
Screenshot 2023-05-31 at 11 23 53 AM
Screenshot 2023-05-31 at 11 20 47 AM

**Comment by Lunaliar** [comment]
Awesome! If I had to take a guess based on the error message "onsole is not defined" there may be a typo on a console log somewhere. Dont worry Ill do a sweep of all console logs before we merge.

File: src/cronjobs/userProfileJobs.js [hunk 1]
```diff
@@ -1,26 +1,26 @@
-const { CronJob } = require('cron');
-const moment = require('moment-timezone');
+const { CronJob } = require("cron");
```
Comment (lchua2314) [inline src/cronjobs/userProfileJobs.js:1]: Please revert the double quotes back to single quotes. I'm assuming it's the linter that's doing this.

File: src/helpers/userHelper.js [hunk 2]
```diff
@@ -1,73 +1,74 @@
...
+const dashboardHelper = require("./dashboardhelper")();
+const reportHelper = require("./reporthelper")();
 
-const emailSender = require('../utilities/emailSender');
+const emailSender = require("../utilities/emailSender");
 
-const logger = require('../startup/logger');
-const hasPermission = require('../utilities/permissions');
+const logger = require("../startup/logger");
+const hasPermission = require("../utilities/permissions");
```
Comment (lchua2314) [inline src/helpers/userHelper.js:14]: Please remove hasPermission because it's an unused constant.

File: src/helpers/userHelper.js [hunk 3]
```diff
@@ -692,75 +713,83 @@ const userHelper = function () {
...
   };
 
   const increaseBadgeCount = async function (personId, badgeId) {
-    console.log('Increase Badge Count', personId, badgeId);
-    userProfile.updateOne({ _id: personId, 'badgeCollection.badge': badgeId },
-    { $inc: { 'badgeCollection.$.count': 1 }, $set: { 'badgeCollection.$.lastModified': Date.now().toString() } , $push:{'badgeCollection.$.earnedDate':earnedDateBadge() }},
-    (err) => {
-      if (err) {
-        console.log(err);
+    console.log("Increase Badge Count", personId, badgeId);
```
Comment (lchua2314) [inline src/helpers/userHelper.js:753]: As chen as stated, please remove console.logs used for testing because they hinder performance.

**Review by lchua2314 — CHANGES_REQUESTED** [review]
Hi, @Lunaliar I want to say again, I really like your video. I have requested some changes that you are familiar with from PR #366: 
1. Reverting double quotes back to single quotes to match the rest of the codebase.
2. hasPermission should be removed because it's an unused constant.
3. As chen has stated, console logs should be removed.

When I tested the feature, I refreshed the profile page and found two duplicate badges. 
Dup badges

When I refresh the page, it shows one badge. 
Refresh

Maybe this is just a frontend-issue or I missed something, but I thought I should bring it up.

Other than that, everything works as intended. Good job!

**Pull request closed without merging**